File: Data Structures & Algorithms/permutations/submission-5.java
class Solution {
    List<List<Integer>> res;
    public List<List<Integer>> permute(int[] nums) {
        res = new ArrayList<>();
        res.add(new ArrayList<>());

        for(int num: nums){
            List<List<Integer>> new_list = new ArrayList<>();
            for(List<Integer> p:res){
                for(int i=0; i<=p.size(); i++){
                    List<Integer> copy = new ArrayList<>(p);
                    copy.add(i, num);
                    new_list.add(copy);
                }
            }

            res = new_list;
        }

        return res;
    }
}

File: Data Structures & Algorithms/subsets/submission-5.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        
        res = [[]]
        
        for n in nums:
            size = len(res)
            for i in range(size):
                subset = res[i].copy()
                subset.append(n)
                res.append(subset)

        return res
        